### PR TITLE
more fixes

### PR DESCRIPTION
**Pull request by yueyinqiu**

### CODE REVIEW

**Comment by NiklasGustafsson** [comment]
I love the energy you're putting into this, but please provide a description of what the PR is for.

File: src/TorchSharp/DataLoader.cs [hunk 1]
```diff
@@ -412,35 +411,42 @@ public bool MoveNext()
                 {
                     DisposeCurrent();
 
-                    using (var scope = torch.NewDisposeScope()) {
```
Comment (yueyinqiu) [inline src/TorchSharp/DataLoader.cs:315]: I just realized that the dispose scope does not work on other threads. Instead dispose scopes should be separately created in the below `ForAll`.

File: src/TorchSharp/DisposeScope.cs [hunk 2]
```diff
@@ -40,7 +40,7 @@ internal DisposeScope(DisposeScopeManager disposeScopeManager)
         public IReadOnlyList<IDisposable> DisposablesView {
             get {
                 if (this._disposeScopeManager is null)
-                    throw new ObjectDisposedException("The dispose scope has been disposed.");
+                    throw new ObjectDisposedException(this.GetType().FullName);
```
Comment (yueyinqiu) [inline src/TorchSharp/DisposeScope.cs:43]: 3. The parameter of the constructor of `ObjectDisposedException` is not the message, but is something called objectName.

File: docfx/articles/memory.md [hunk 3]
```diff
@@ -295,6 +295,107 @@ public override Tensor entropy() => torch.WrappedTensorDisposeScope(() => ((scal
 
 ```
 
+## Data loaders and datasets
```
Comment (yueyinqiu) [inline docfx/articles/memory.md:298]: 4. add some document about data loaders and datasets.

**Comment by yueyinqiu** [comment]
@NiklasGustafsson 

I have added some comments on the codes. (4 in total)

File: src/TorchSharp/Tensor/Tensor.cs [hunk 4]
```diff
@@ -7377,6 +7377,12 @@ public static long max_int_value(ScalarType type)
         public static DisposeScope NewDisposeScope() =>
             DisposeScopeManager.ThreadSingleton.NewDisposeScope();
 
+        /// <summary>
+        /// Get the current dispose scope for the current thread.
+        /// </summary>
+        public static DisposeScope? CurrentDisposeScope =>
```
Comment (yueyinqiu) [inline src/TorchSharp/Tensor/Tensor.cs:7383]: 5. expose the current dispose scope and add Attach method for ienumerable, which could be used with the not-disposeBatch data loaders to manually attach the tensors to the current scope

**Comment by NiklasGustafsson** [comment]
@yueyinqiu -- ready to merge, or are there more changes coming?

**Comment by yueyinqiu** [comment]
I think it's ready

**Pull request merged**